core/src/components/item-divider/test/spec/item-divider.e2e.ts

```diff
@@ -11,4 +11,21 @@ test.describe('item-divider: spec', () => {
 
     await expect(page).toHaveScreenshot(`list-item-divider-${page.getSnapshotSettings()}.png`);
   });
+
+  test('should have margin-end on button in end slot', async ({ page, skip }) => {
+    skip.mode('ios', 'This behavior is only available in MD mode.');
+
+    test.info().annotations.push({
+      type: 'issue',
+      description: 'https://github.com/ionic-team/ionic-framework/issues/17012',
+    });
+
+    const button = page.locator('ion-button');
+    const buttonMarginEnd = await button.evaluate((el) => {
+      const styles = window.getComputedStyle(el);
+      return styles.getPropertyValue('margin-end');
+    });
+
+    expect(buttonMarginEnd).not.toBe('0px');
```

Comment:
We should avoid checking exact pixel values in favor of screenshot tests. It looks like test coverage for buttons in dividers was added in https://github.com/ionic-team/ionic-framework/commit/426913d0de50e65a76b029258daa09e6567c6515, so I'm not sure we even need this test.